modules/light-clients/06-solomachine/light_client_module_test.go

```diff
@@ -1,18 +1,797 @@
+					suite.Require().NoError(err)
+					signBytes = solomachine.SignBytes{
+						Sequence:    sm.Sequence,
+						Timestamp:   sm.Time,
+						Diversifier: sm.Diversifier,
+						Path:        key,
+						Data:        commitmentBz,
+					}
+
+					signBz, err := suite.chainA.Codec.Marshal(&signBytes)
+					suite.Require().NoError(err)
+
+					sig := sm.GenerateSignature(signBz)
+
+					signatureDoc := &solomachine.TimestampedSignatureData{
+						SignatureData: sig,
+						Timestamp:     sm.Time,
+					}
+
+					proof, err = suite.chainA.Codec.Marshal(signatureDoc)
+					suite.Require().NoError(err)
+				},
+				nil,
+			},
+			{
+				"success: packet acknowledgement verification",
+				func() {
+					path = sm.GetPacketAcknowledgementPath(ibctesting.MockPort, ibctesting.FirstChannelID, 1)
+					merklePath, ok := path.(commitmenttypes.MerklePath)
+					suite.Require().True(ok)
+					key, err := merklePath.GetKey(1) // index 0 is the key for the IBC store in the multistore, index 1 is the key in the IBC store
+					suite.Require().NoError(err)
+					signBytes = solomachine.SignBytes{
+						Sequence:    sm.Sequence,
+						Timestamp:   sm.Time,
+						Diversifier: sm.Diversifier,
+						Path:        key,
+						Data:        ibctesting.MockAcknowledgement,
+					}
+
+					signBz, err := suite.chainA.Codec.Marshal(&signBytes)
+					suite.Require().NoError(err)
+
+					sig := sm.GenerateSignature(signBz)
+
+					signatureDoc := &solomachine.TimestampedSignatureData{
+						SignatureData: sig,
+						Timestamp:     sm.Time,
+					}
+
+					proof, err = suite.chainA.Codec.Marshal(signatureDoc)
+					suite.Require().NoError(err)
+				},
+				nil,
+			},
+			{
+				"success: packet receipt verification",
+				func() {
+					path = sm.GetPacketReceiptPath(ibctesting.MockPort, ibctesting.FirstChannelID, 1)
+					merklePath, ok := path.(commitmenttypes.MerklePath)
+					suite.Require().True(ok)
+					key, err := merklePath.GetKey(1) // in a multistore context: index 0 is the key for the IBC store in the multistore, index 1 is the key in the IBC store
+					suite.Require().NoError(err)
+					signBytes = solomachine.SignBytes{
+						Sequence:    sm.Sequence,
+						Timestamp:   sm.Time,
+						Diversifier: sm.Diversifier,
+						Path:        key,
+						Data:        []byte{byte(1)}, // packet receipt is stored as a single byte
+					}
+
+					signBz, err := suite.chainA.Codec.Marshal(&signBytes)
+					suite.Require().NoError(err)
+
+					sig := sm.GenerateSignature(signBz)
+
+					signatureDoc := &solomachine.TimestampedSignatureData{
+						SignatureData: sig,
+						Timestamp:     sm.Time,
+					}
+
+					proof, err = suite.chainA.Codec.Marshal(signatureDoc)
+					suite.Require().NoError(err)
+				},
+				nil,
+			},
+			// TODO: Cov missing, need a more extensive refactor of tests in order to cov it.
+			// {
+			// 	"client not found",
+			// 	func() {},
+			// 	clienttypes.ErrClientNotFound,
+			// },
```

Comment:
Isn't it possible to test this by setting `clientID` to some value for which there isn't any client store?